dashboards/grafana/kubevirt-control-plane.json

```diff
@@ -1,3087 +1,3640 @@
-                "total": false,
-                "values": true
-            },
-            "lines": true,
-            "linewidth": 1,
-            "links": [],
-            "nullPointMode": "null",
-            "options": {
-                "alertThreshold": true
-            },
-            "percentage": false,
-            "pluginVersion": "7.5.4",
-            "pointradius": 5,
-            "points": false,
-            "renderer": "flot",
-            "repeat": null,
-            "seriesOverrides": [],
-            "spaceLength": 10,
-            "stack": true,
-            "steppedLine": false,
-            "targets": [
-                {
-                    "exemplar": true,
-                    "expr": "sum(rate(process_open_fds{container!=\"\"}[10m])) by (container)",
-                    "format": "time_series",
-                    "interval": "",
-                    "intervalFactor": 2,
-                    "legendFormat": "{{container}}",
-                    "refId": "A"
-                }
-            ],
-            "thresholds": [],
-            "timeFrom": null,
-            "timeRegions": [],
-            "timeShift": null,
-            "title": "Open Files",
-            "tooltip": {
-                "shared": false,
-                "sort": 0,
-                "value_type": "individual"
-            },
-            "type": "graph",
-            "xaxis": {
-                "buckets": null,
-                "mode": "time",
-                "name": null,
-                "show": true,
-                "values": []
+          "aliasColors": {},
+          "bars": false,
+          "dashLength": 10,
+          "dashes": false,
+          "datasource": "${datasource}",
+          "fill": 7,
+          "fillGradient": 0,
+          "gridPos": {
+            "h": 8,
+            "w": 8,
+            "x": 0,
+            "y": 20
+          },
+          "hiddenSeries": false,
+          "id": 24,
+          "legend": {
+            "alignAsTable": true,
+            "avg": false,
+            "current": false,
+            "hideEmpty": true,
+            "hideZero": true,
+            "max": true,
+            "min": false,
+            "rightSide": true,
+            "show": true,
+            "sort": "max",
+            "sortDesc": true,
+            "total": false,
+            "values": true
+          },
+          "lines": true,
+          "linewidth": 1,
+          "nullPointMode": "null",
+          "options": {
+            "alertThreshold": true
+          },
+          "percentage": false,
+          "pluginVersion": "8.2.3",
+          "pointradius": 2,
+          "points": false,
+          "renderer": "flot",
+          "seriesOverrides": [],
+          "spaceLength": 10,
+          "stack": true,
+          "steppedLine": false,
+          "targets": [
+            {
+              "exemplar": true,
+              "expr": "kubevirt_workqueue_unfinished_work_seconds{job=~\".*kubevirt.*\", instance=~\"$instance\"}",
```

Comment:
The [documentation](https://github.com/kube-reporting/metering-operator/blob/master/pkg/util/workqueue/prometheus/prometheus.go#L107) for this metric says: "One can deduce the number of stuck threads by observing the rate at which this increases." 

So that's why I used the `increase()` function, since we need to check the rate of this metric to understand the result....

Why do you think we should not use a rate function in this query? 
obs: I've no problem changing it, but I need to understand why...

Reply:
`rate()` (and therefore, `increase()`) functions are meant to be used with monotonic metrics and aren't suitable for gauges such as `kubevirt_workqueue_unfinished_work_seconds`. Here's an example of graphing this metric with and without the `sum(increase())`. The data is collected from repeated runs of a 10-VMI creations + deletions:

### With `sum(increase())`:

![2021-11-21 13_39_30-KubeVirt _ Control Plane - Grafana — Mozilla Firefox](https://user-images.githubusercontent.com/17428696/142760550-dffd1c83-54e9-4f26-817e-50600026c090.png)

### Without `sum(increase())`:

![2021-11-21 13_38_18-KubeVirt _ Control Plane - Grafana — Mozilla Firefox](https://user-images.githubusercontent.com/17428696/142760593-c172f1de-4d40-46af-96fa-bf9bc7400cd9.png)

IMO, the 1st form (with `sum(increase())`) portrays a misleading picture where unfinished work had a relatively large value at every single moment during this simulation, and partially (though not entirely) blurs the bursty nature of the simulation. This happens because the `sum(increase(...{}[5m]))` runs over a sliding window of 5m and tends to smooth it. 

So, I think that the 2nd form provides a better view of what happens, and by visually inspecting it we can definitely observe the increase of the metric and deduce about stuck threads (by the way it seems we have stuck or just very slow threads in virt-handler ;-) )

Reply:
Humm good catch, those are Gauge metrics and should not be used with `rate()` and `increase()`.